Cesium3DTilesSelection/src/EllipsoidTilesetLoader.cpp

```diff
@@ -0,0 +1,274 @@
+  const GlobeRectangle& globeRectangle = boundingRegion.getRectangle();
+
+  Tile& child = children.emplace_back(parent.getLoader());
+  child.setTileID(childID);
+  child.setRefine(parent.getRefine());
+  child.setTransform(glm::translate(
+      glm::dmat4x4(1.0),
+      _projection.getEllipsoid().cartographicToCartesian(
+          globeRectangle.getNorthwest())));
+  child.setBoundingVolume(boundingRegion);
+  child.setGeometricError(
+      6.0 * calcQuadtreeMaxGeometricError(_projection.getEllipsoid()) *
+      globeRectangle.computeWidth());
+}
+
+BoundingRegion EllipsoidTilesetLoader::createBoundingRegion(
+    const QuadtreeTileID& quadtreeID) const {
+  return BoundingRegion(
+      _projection.unproject(_tilingScheme.tileToRectangle(quadtreeID)),
+      0.0,
+      0.0,
+      _projection.getEllipsoid());
+}
+
+EllipsoidTilesetLoader::Geometry
+EllipsoidTilesetLoader::createGeometry(const Tile& tile) const {
+  static constexpr uint16_t resolution = 24;
+
+  std::vector<uint16_t> indices(6 * (resolution - 1) * (resolution - 1));
+  std::vector<glm::vec3> vertices(resolution * resolution);
+  std::vector<glm::vec3> normals(vertices.size());
+
+  const Ellipsoid& ellipsoid = _projection.getEllipsoid();
+  const GlobeRectangle& rectangle =
+      std::get<BoundingRegion>(tile.getBoundingVolume()).getRectangle();
+
+  double west = rectangle.getWest();
+  double east = rectangle.getEast();
+  double north = rectangle.getNorth();
+  double south = rectangle.getSouth();
+
+  double lonStep = (east - west) / (resolution - 1);
+  double latStep = (south - north) / (resolution - 1);
+
+  glm::dmat4 inverseTransform = glm::inverse(tile.getTransform());
+
+  for (uint16_t x = 0; x < resolution; x++) {
+    double longitude = (lonStep * x) + west;
+    for (uint16_t y = 0; y < resolution; y++) {
+      double latitude = (latStep * y) + north;
+      Cartographic cartographic(longitude, latitude);
+
+      uint16_t index = static_cast<uint16_t>((resolution * x) + y);
+      vertices[index] = glm::dvec3(
+          inverseTransform *
+          glm::dvec4(ellipsoid.cartographicToCartesian(cartographic), 1.0));
+      normals[index] = ellipsoid.geodeticSurfaceNormal(cartographic);
+
+      if (x < resolution - 1 && y < resolution - 1) {
+        uint16_t a = index + 1;
+        uint16_t b = index + resolution;
+        uint16_t c = b + 1;
+        indices.insert(indices.end(), {b, index, a, b, a, c});
+      }
+    }
+  }
+
+  return Geometry{std::move(indices), std::move(vertices), std::move(normals)};
+}
+
+Model EllipsoidTilesetLoader::createModel(const Geometry& geometry) const {
+  const std::vector<uint16_t>& indices = geometry.indices;
+  const std::vector<glm::vec3>& vertices = geometry.vertices;
+  const std::vector<glm::vec3>& normals = geometry.normals;
+
+  size_t indicesSize = indices.size() * sizeof(uint16_t);
+  size_t verticesSize = vertices.size() * sizeof(glm::vec3);
+  size_t normalsSize = verticesSize;
+
+  Model model;
```

Comment:
Also, you need to add:

```
model.extras["gltfUpAxis"] = JsonValue(std::underlying_type_t<CesiumGeometry::Axis>(CesiumGeometry::Axis::Z));
```

Without that, the glTF is assumed to use a coordinate system in which Y is considered Up, and cesium-native will add an additional transformation to the model to make Z up. It might be hard to tell this is happening with a smooth ellipsoid, but it means the radii at various points on Earth may not be correct.